csrc/device_lower/analysis/tensor_memory.cpp

```diff
@@ -193,7 +198,38 @@ TMemAlllocationInfo computeTMemAlllocationInfo(Fusion* fusion) {
 std::pair<
     std::unordered_map<TensorView*, TMemRegisterDataPath>,
     std::unordered_map<TensorView*, TMemRegisterDataPath>>
-computeTMemLdStDataPath(Fusion* fusion) {
+computeTMemLdStDataPath(Fusion* fusion, const TMemAlllocationInfo& allocation) {
+  // This function uses simplifyExpr extensively. If we have disable expression
+  // simplification in order to help inspect generated kernels then we will get
+  // incorrect results here. Instead, we ensure it is enabled using this guard.
+  DisableOptionsGuard dog;
+  DisableOptionsGuard::getCurOptions().unset(DisableOption::ExprSimplify);
+  // In the CUDA programming model, each CTA has TIDx, TIDy, and TIDz.
+  // Unfortunatly, the mapping of these TIDs to hardware concepts like warp,
+  // warp group, are not clear and depend on the kernel launch configuration.
+  // Here, we try to not assume anything like "TIDx must be a multiple of 32",
+  // but still, we must be able to validate and pattern match the data access
+  // of the tensor memory load/store.
+  const auto& pdim_map = GpuLower::current()->parallelDimensionMap();
+  // Get the TID Parallel types that we are interested in. We ignore parallel
+  // types that are not used in the kernel, and the ones that have size 1.
+  std::vector<ParallelType> tid_ptypes;
```

Comment:
I think it's worth noting this vector is ordered in this particular way since that's important below.

Reply:
Comment added